$MirOS: ports/textproc/gdiff/patches/patch-src_context_c,v 1.1.7.1 2005/03/18 15:52:42 tg Exp $
--- src/context.c.orig	Thu Feb 28 04:22:20 2002
+++ src/context.c	Tue Mar 25 21:51:44 2008
@@ -61,9 +61,9 @@ print_context_label (char const *mark,
       int nsec = TIMESPEC_NS (inf->stat.st_mtim);
       if (! (tm && nstrftime (buf, sizeof buf, time_format, tm, 0, nsec)))
 	{
-	  long sec = inf->stat.st_mtime;
+	  long long sec = inf->stat.st_mtime;
 	  verify (info_preserved, sizeof inf->stat.st_mtime <= sizeof sec);
-	  sprintf (buf, "%ld.%.9d", sec, nsec);
+	  snprintf (buf, sizeof (buf), "%lld.%.9d", sec, nsec);
 	}
       fprintf (outfile, "%s %s\t%s\n", mark, inf->name, buf);
     }
